### PR TITLE
Serialize trajectory node data.

### PR DESCRIPTION
**Pull request by wohe**
Related to #253.

### CODE REVIEW

File: cartographer/mapping/proto/serialization.proto [hunk 1]
```diff
@@ -18,21 +18,21 @@ package cartographer.mapping.proto;
...
+message NodeData {
   optional NodeId node_id = 1;
-  optional sensor.proto.CompressedRangeData range_data = 4;
+  optional TrajectoryNode trajectory_node = 5;
 }
 
 message SerializedData {
   optional Submap submap = 1;
-  optional RangeData range_data = 2;
+  optional NodeData node_data = 2;
```
Comment (cschuet) [inline cartographer/mapping/proto/serialization.proto]: Is it fine to break proto backwards compatibility? You could also add [deprecated=true] to range_data.

File: cartographer/mapping/trajectory_node.h [hunk 2]
```diff
@@ -56,6 +57,8 @@ struct TrajectoryNode {
   transform::Rigid3d pose;
 };
 
+proto::TrajectoryNode ToProto(const TrajectoryNode::Data& constant_data);
```
Comment (cschuet) [inline cartographer/mapping/trajectory_node.h]: I don't see FromProto. Where are you planning to deserialize the data? Add FromProto and a round-trip unit test?

File: cartographer/mapping/proto/serialization.proto (hunk 1 above)
Comment (wohe) [reply cartographer/mapping/proto/serialization.proto]: Yes. We are before version 1.0 and make no guarantees.

**Comment by ojura** [comment]
Please have a look at #508.

**Pull request merged**